pkl-core/src/main/java/org/pkl/core/service/ExecutorSpiImpl.java

```diff
@@ -98,4 +128,61 @@ public String evaluatePath(Path modulePath, ExecutorSpiOptions options) {
       ModuleKeyFactories.closeQuietly(builder.getModuleKeyFactories());
     }
   }
+
+  private HttpClient getOrCreateHttpClient(ExecutorSpiOptions options) {
+    List<Path> certificateFiles;
+    List<URI> certificateUris;
+    if (options instanceof ExecutorSpiOptions2) {
+      var options2 = (ExecutorSpiOptions2) options;
+      certificateFiles = options2.getCertificateFiles();
+      certificateUris = options2.getCertificateUris();
+    } else {
```

Comment:
To handle if pkl-executor is older than this distribution:

```suggestion
    try {
      if (options instanceof ExecutorSpiOptions2) {
        var options2 = (ExecutorSpiOptions2) options;
        certificateFiles = options2.getCertificateFiles();
        certificateUris = options2.getCertificateUris();
      } else {
        certificateFiles = List.of();
        certificateUris = List.of();
      }
    } catch (NoClassDefFoundError e) {
```

And we can enable the test if the executor is older than the distribution version.

Reply:
This is really testing the limits, and I can't actually get it to work. I think it's far safer to ship `pkl-executor` with `pkl-config-java-all` or some other distribution.

Reply:
We can probably find a solution that won’t require including pkl-executor in distributions. Until then, we can leave the “older pkl-executor, newer distribution” test disabled.

Reply:
This change is passing for me:

```diff
diff --git a/pkl-core/src/main/java/org/pkl/core/service/ExecutorSpiImpl.java b/pkl-core/src/main/java/org/pkl/core/service/ExecutorSpiImpl.java
index 3af2df3eb..1c713cb76 100644
--- a/pkl-core/src/main/java/org/pkl/core/service/ExecutorSpiImpl.java
+++ b/pkl-core/src/main/java/org/pkl/core/service/ExecutorSpiImpl.java
@@ -132,11 +132,16 @@ public class ExecutorSpiImpl implements ExecutorSpi {
   private HttpClient getOrCreateHttpClient(ExecutorSpiOptions options) {
     List<Path> certificateFiles;
     List<URI> certificateUris;
-    if (options instanceof ExecutorSpiOptions2) {
-      var options2 = (ExecutorSpiOptions2) options;
-      certificateFiles = options2.getCertificateFiles();
-      certificateUris = options2.getCertificateUris();
-    } else {
+    try {
+      if (options instanceof ExecutorSpiOptions2) {
+        var options2 = (ExecutorSpiOptions2) options;
+        certificateFiles = options2.getCertificateFiles();
+        certificateUris = options2.getCertificateUris();
+      } else {
+        certificateFiles = List.of();
+        certificateUris = List.of();
+      }
+    } catch (NoClassDefFoundError e) {
       certificateFiles = List.of();
       certificateUris = List.of();
     }
diff --git a/pkl-executor/src/test/kotlin/org/pkl/executor/EmbeddedExecutorTest.kt b/pkl-executor/src/test/kotlin/org/pkl/executor/EmbeddedExecutorTest.kt
index 3658b3ead..46e014274 100644
--- a/pkl-executor/src/test/kotlin/org/pkl/executor/EmbeddedExecutorTest.kt
+++ b/pkl-executor/src/test/kotlin/org/pkl/executor/EmbeddedExecutorTest.kt
@@ -39,7 +39,7 @@ class EmbeddedExecutorTest {
 
         // This context has a pkl-executor version that is lower than the distribution version.
         // It can be enabled once there is a distribution that includes pkl-executor.
-        //ExecutionContext(executor1_2.value, ::convertToOptions1, "Options1, Executor1, Distribution2"),
+        ExecutionContext(executor1_2.value, ::convertToOptions1, "Options1, Executor1, Distribution2"),
 
         ExecutionContext(executor2_1.value, ::convertToOptions1, "Options1, Executor2, Distribution1"),
         ExecutionContext(executor2_1.value, ::convertToOptions2, "Options2, Executor2, Distribution1"),
```

Reply:
Didn't work for me, but maybe IntelliJ didn't pick up my code change (I've been struggling with this a lot). I don't know if `NoClassDefFoundError` is guaranteed to be only thrown here though. It's a fairly extreme solution.